infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -498,41 +620,53 @@ public void updateSnapshotSyncStatusCompleted(String clusterId, long remainingEn
      *
      * Note: TransactionAbortedException has been handled by upper level.
      *
-     * @param clusterId sink cluster id
+     * @param session session with the remote sink cluster
      */
-    public void updateSyncStatus(String clusterId, SyncType lastSyncType, SyncStatus status) {
-        ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setClusterId(clusterId).build();
+    public void updateSyncStatus(LogReplicationSession session, SyncType lastSyncType, SyncStatus status) {
 
         try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
 
-            CorfuStoreEntry<ReplicationStatusKey, ReplicationStatusVal, Message> record = txn.getRecord(replicationStatusTable, key);
+            CorfuStoreEntry<LogReplicationSession, ReplicationStatus, Message> entry = txn.getRecord(statusTable,
+                session);
 
             // When a remote cluster has been removed from topology, the corresponding entry in the status table is
             // removed and FSM is shutdown. Since FSM shutdown is async, we ensure that we don't update a record which
             // has already been deleted.
             // (STOPPED status is used for other FSM states as well, so cannot rely only on the incoming status)
-            if(record.getPayload() == null && status == SyncStatus.STOPPED) {
-                log.debug("syncStatus :: ignoring update for {} to syncType {} and status {} as no record exists for the same",
-                        clusterId, lastSyncType, status);
+            if (entry.getPayload() == null && status == SyncStatus.STOPPED) {
+                log.debug("syncStatus :: ignoring update for session {} to syncType {} and status {} as no record " +
+                    "exists for the same", session, lastSyncType, status);
```

Comment:
can we add a `txn.commit()` here since this is an expected path to complete the transaction?